src/openMVG/voctree/descriptor_loader.cpp

```diff
@@ -55,6 +56,20 @@ void getListOfDescriptorFiles(const std::string &fileFullPath, std::map<IndexT,
 
   bfs::path bp(fileFullPath);
 
+  // If the input is a directory, list all .desc files recursively.
+  if(bfs::is_directory(bp))
+  {
+    std::size_t viewId = 0;
+    for(bfs::recursive_directory_iterator it(bp), end; it != end; ++it)
+    {
+        if(!bfs::is_directory(*it) && boost::algorithm::ends_with(it->path().string(), ".desc"))
```

Comment:
easier: `it->path().extension()==".desc"`
http://www.boost.org/doc/libs/1_60_0/libs/filesystem/doc/reference.html#path-extension